usage/bigquery.go

```diff
@@ -82,23 +91,59 @@ func (bq *bigqueryHandler) QueryUsageSummary(ctx context.Context, userID string,
 	return &bqRows[0], nil
 }
 
+func (bq *bigqueryHandler) QueryUsageSummaryWithTimestep(ctx context.Context, userID string, creatorID string, spec QuerySpec) (*[]UsageSummaryRow, error) {
+	sql, args, err := buildUsageSummaryQuery(bq.opts.HourlyUsageTable, userID, creatorID, spec)
+	if err != nil {
+		return nil, fmt.Errorf("error building usage summary query: %w", err)
+	}
+
+	bqRows, err := doBigQuery[UsageSummaryRow](bq, ctx, sql, args)
+	if err != nil {
+		return nil, fmt.Errorf("bigquery error: %w", err)
+	}
+
+	if len(bqRows) == 0 {
+		return nil, nil
+	}
+
+	return &bqRows, nil
+}
+
 func buildUsageSummaryQuery(table string, userID string, creatorID string, spec QuerySpec) (string, []interface{}, error) {
 	if userID == "" {
 		return "", nil, fmt.Errorf("userID cannot be empty")
 	}
 
 	query := squirrel.Select(
-		"cast(sum(transcode_total_usage_minutes) as FLOAT64) as transcode_total_usage_minutes",
+		"cast(sum(transcode_total_usage_mins) as FLOAT64) as transcode_total_usage_mins",
 		"cast(sum(delivery_usage_gbs) as FLOAT64) as delivery_usage_gbs",
-		"cast(avg(storage_usage_gbs) as FLOAT64) as storage_usage_gbs").
+		"cast((sum(storage_usage_mins) / count(distinct usage_hour_ts)) as FLOAT64) as storage_usage_mins").
 		From(table).
-		Limit(2)
+		Limit(maxBigQueryResultRows)
```

Comment:
I think we should make like the views query and make an explicit check for when there are more results than we allow. Otherwise callers can get 10k entries in the reponse and have no idea if that's incomplete data or that it actually got cropped cause their query is too granular.

The main changes are just changing this limit to `max+1`+ instead then checking if the response count is `>max`, returning an error if that's the case.